src/poc/miner_poc_statem.erl

```diff
@@ -793,6 +799,20 @@ send_onion(P2P, Onion, Retry) ->
             send_onion(P2P, Onion, Retry-1)
     end.
 
+is_self_witness(WitnessPubkeyBin, Challengees, Ledger) ->
+    %% We check for the following:
+    %% - Are we beacon only?
+    %% - If yes, is there are single challengee? (with beacon only, there can only be ONE challengee)
+    %% - If yes, is the challengee the same as this witness, i.e. self-witnessing
+    case blockchain:config(?poc_path_limit, Ledger) of
+        {ok, 1} ->
```

Comment:
I'm not sure I like tying these together? maybe just something like `allow_self_witness`?